sklearn/decomposition/tests/test_fastica.py

```diff
@@ -223,20 +250,26 @@ def test_non_square_fastica(add_noise):
 
     # Check that we have estimated the original sources
     if not add_noise:
-        assert_almost_equal(np.dot(s1_, s1) / n_samples, 1, decimal=3)
-        assert_almost_equal(np.dot(s2_, s2) / n_samples, 1, decimal=3)
+        assert_allclose(np.dot(s1_, s1) / n_samples, 1, atol=1e-3)
+        assert_allclose(np.dot(s2_, s2) / n_samples, 1, atol=1e-3)
 
 
-def test_fit_transform():
+def test_fit_transform(global_random_seed):
     """Test unit variance of transformed data using FastICA algorithm.
 
     Check that `fit_transform` gives the same result as applying
     `fit` and then `transform`.
 
     Bug #13056
     """
-    rng = np.random.RandomState(0)
+    # XXX: casting X with `.astype(global_dtype)` reveals a rare numerical
+    # stability problem of FastICA with float32 bit data. I can triggers
+    # NaN values somewhere in the computation, only for some random seeds.
```

Comment:
Note to reviewers: I did not enable `global_dtype` on this test because that would make it fail for some random seeds by revealing a numerical stability problem of our fast ICA  implementation on 32 bit data.

Reply:
I am investigating a bit, and this can happen for instance by setting `SKLEARN_TESTS_GLOBAL_RANDOM_SEED=60` and float32.

But I noticed that the model does not converge in float64 for that same seed while it generally does converge for other seeds.

Reply:
I fixed the NaN problem in [4464f9d](https://github.com/scikit-learn/scikit-learn/pull/22806/commits/4464f9d4d77d462f845f1048e81283ca8ca9ff4b) as explained in https://github.com/scikit-learn/scikit-learn/pull/22806#discussion_r831453093.